### PR TITLE
[WIP] Preload fair content from a local web server advertised through Bonjour.

### PR DESCRIPTION
**Pull request by alloy**
See #186.

### CODE REVIEW

**Comment by alloy** [comment]
Oops, looks like I left in a commit that was meant to be squashed, will do that tomorrow.

**Comment by orta** [comment]
lols FairEnough

File: Artsy Tests/ARFairContentPreloaderTests.m [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+        if (!preloader.hasResolvedService) {
+            preloader = nil;
+            // TODO how can we best fail all the test cases in this group?
+        } else {
+
+        }
+    });
+
+    it(@"fetches the manifest", ^{
+        [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *request) {
```
Comment (orta) [inline Artsy Tests/ARFairContentPreloaderTests.m]: I'm pretty sure we have some categories to make that a one liner

File: Artsy Tests/ARFairContentPreloaderTests.m [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+                                                    statusCode:200
+                                                       headers:@{ @"Content-Type":@"application/json" }];
+        }];
+
+        __block BOOL fetched = NO;
+        [preloader fetchManifest:^(id _) { fetched = YES; }];
+        while (!fetched) {
+            CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0.25, true);
+        }
+        expect(preloader.manifest).to.equal(manifest);
```
Comment (orta) [inline Artsy Tests/ARFairContentPreloaderTests.m]: ^ the pattern above can be replicated via `expect(fetched).will.betruth();`

File: Artsy Tests/ARFairContentPreloaderTests.m (hunk 1 above)
Comment (alloy) [inline Artsy Tests/ARFairContentPreloaderTests.m]: Aren’t those using Artsy base URLs?

File: Artsy Tests/ARFairContentPreloaderTests.m (hunk 2 above)
Comment (alloy) [inline Artsy Tests/ARFairContentPreloaderTests.m]: Sweet :+1:

Comment (alloy) [inline Artsy Tests/ARFairContentPreloaderTests.m]: I have been trying to use this, but have been unsuccessful. For some reason it seems to be unreliable, which might be related to the CI issues we’ve been having. I’ll just leave these snippets in for now.

**Comment by orta** [comment]
been wondering about annotations on routes that could say "this can run through FairEnough", can discuss at https://github.com/artsy/mobile/issues/22

**Comment by orta** [comment]
worth keeping this open?

**Comment by alloy** [comment]
Maybe not the PR, no, but I’d like to keep the branch.

**Pull request closed without merging**